File: CppPractice/CppLeetcode/Leetcode145.cpp
#include<vector>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode():val(0),left(nullptr),right(nullptr){};
    TreeNode(int x):val(x),left(nullptr),right(nullptr){};
    TreeNode(int x, TreeNode *leftp,TreeNode *rightp):val(x),left(leftp),right(rightp){};
};

class Solution {
private:
    void Traversal(TreeNode *p,vector<int> &res)
    {
        if (!p)
        {
            return;
        }
        else
        {
            Traversal(p->left,res);
            Traversal(p->right,res);
            res.push_back(p->val);
        }
    }
public:
    vector<int> postorderTraversal(TreeNode* root) 
    {
        vector<int> result;
        Traversal(root,result);
        return result;
    }
};

File: Pythonleetcode/Leetcode145.py
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution(object):
    def helper(self, root, res):
        if not root:
            return
        self.helper(root.left, res)
        self.helper(root.right, res)
        res.append(root.val)

    def postorderTraversal(self, root):
        res = []
        self.helper(root, res)
        return res


